arduino/cores/packagemanager/loader.go

```diff
@@ -374,14 +374,39 @@ func (pm *PackageManager) loadBoards(platform *cores.PlatformRelease) error {
 
 	propertiesByBoard := boardsProperties.FirstLevelOf()
 
-	platform.Menus = propertiesByBoard["menu"]
+	if menus, ok := propertiesByBoard["menu"]; ok {
+		platform.Menus = menus
+	} else {
+		platform.Menus = properties.NewMap()
+	}
 
-	delete(propertiesByBoard, "menu") // TODO: check this one
+	// This is not a board id so we remove it to correctly
+	// set all other boards properties
+	delete(propertiesByBoard, "menu")
 
+	skippedBoards := []string{}
 	for boardID, boardProperties := range propertiesByBoard {
-		boardProperties.Set("_id", boardID) // TODO: What is that for??
-		board := platform.GetOrCreateBoard(boardID)
+		var board *cores.Board
+		for _, key := range boardProperties.FirstLevelKeys() {
+			if key == "menu" && platform.Menus.Size() == 0 {
+				fqbn := fmt.Sprintf("%s:%s:%s", platform.Platform.Package.Name, platform.Platform.Architecture, boardID)
+				skippedBoards = append(skippedBoards, fqbn)
+				goto next_board
+			}
+		}
+		// The board's ID must be available in a board's properties since it can
+		// be used in all configuration files for several reasons, like settings compilation
+		// flags depending on the board id.
+		// For more info:
+		// https://arduino.github.io/arduino-cli/latest/platform-specification/#global-predefined-properties
+		boardProperties.Set("_id", boardID)
+		board = platform.GetOrCreateBoard(boardID)
 		board.Properties.Merge(boardProperties)
+	next_board:
+	}
+
+	if len(skippedBoards) > 0 {
+		return fmt.Errorf("skip loading of boards %s: malformed custom board options", strings.Join(skippedBoards, ", "))
```

Comment:
```suggestion
		return fmt.Errorf("skipping loading of boards %s: malformed custom board options", strings.Join(skippedBoards, ", "))
```
"skip" might give the user the impression that they are being given instructions to do something.

I'm on the fence between "skipped" and "skipping", but I guess "skipping" matches better with the "loading platform release ... : loading boards: ..." context it occurs in?